mozetl/taar/taar_amodump.py

```diff
@@ -104,130 +73,113 @@ class AMOAddonInfo(JSONSchema):
             'ratings': typing.Dict[str, float],
             'summary': typing.Dict[str, str],
             'tags': typing.List[str],
-            'weekly_downloads': int
-    }
-
-
-class URLIteratorFactory(object):
-    BATCH_SIZE = 500
-
-    def __init__(self, cache_dir):
-        self._fnames = os.listdir(cache_dir)
-        logger.info("Processing %d GUIDs" % len(self._fnames))
-
-    def generatorFactory(self):
-        data_slice, self._fnames = self._fnames[:self.BATCH_SIZE], self._fnames[self.BATCH_SIZE:]
-        logger.info("%d in current batch. %d remaining URLs" % (len(data_slice), len(self._fnames)))
-        for i, fname in enumerate(data_slice):
-            guid = fname.split(".json")[0]
-            url = "https://addons.mozilla.org/api/v3/addons/addon/%s/versions/" % guid
-            yield url
+            'weekly_downloads': int}
 
 
 class AMODatabase:
-    def __init__(self, root_path, worker_count):
+    def __init__(self, worker_count):
         """
         Just setup the page_count
         """
-
-        self._root_path = root_path
-        logger.info("Writing temp files to : %s" % self._root_path)
         self._max_processes = worker_count
 
-        self._amo_cache_dir = os.path.join(self._root_path, "amo_cache")
-        self._amo_date_cache_dir = os.path.join(self._root_path, "amo_cache_dates")
-
         uri = DEFAULT_AMO_REQUEST_URI + QUERY_PARAMS
         response = requests.get(uri)
         jdata = json.loads(response.content.decode('utf8'))
 
         self._page_count = jdata['page_count']
 
-    def fetch_pages(self):
-        with guid_cache(self._amo_cache_dir) as db:
-            urls = []
-            for i in range(1, self._page_count+1):
-                url = "%s%s%s" % (DEFAULT_AMO_REQUEST_URI, QUERY_PARAMS, ("&page=%d" % i))
-                urls.append(url)
-            logger.info("Processing AMO urls")
-            p = pool.Pool.from_urls(urls, num_processes=self._max_processes)
+    def fetch_addons(self):
+        addon_map = self._fetch_pages()
+        self._fetch_versions(addon_map)
+        final_result = {}
+        for k, v in addon_map.items():
+            if 'first_create_date' in v:
+                final_result[k] = v
+        logger.info("Final addon set includes %d addons." % len(final_result))
+        return final_result
+
+    def _fetch_pages(self):
+        addon_map = {}
+
+        urls = []
+        for i in range(1, self._page_count+1):
+            url = "%s%s%s" % (DEFAULT_AMO_REQUEST_URI, QUERY_PARAMS, ("&page=%d" % i))
```

Comment:
I know this existed in the previous version too, so not a big deal. String formatting would look better here: 

```python
url = "{}{}{}&page={}".format(...)
```